pkg/util/vralloc/sharedalloc.go

```diff
@@ -0,0 +1,124 @@
+// Licensed to the LF AI & Data foundation under one
+// or more contributor license agreements. See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership. The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package vralloc
+
+type SharedAllocator struct {
```

Comment:
I think `SharedAllocator` is not needed and we can just put `parent` field to `GroupedAllocator`

Reply:
Grouped allocator is originally designed as an internal node in the allocator tree, on which the allocate and release operations are not allowed. I suppose it would be beneficial if we differentiate them like that.